Unknown type name 'wchar_t' in stubs

Step to reproduce:

- Open `coreutils` project
- Choose `libcoreutils.a` target
- Generate tests for `coreutils/gnulib/lib/malloc/dynarray_at_failure.c`
- See error in logs
```
2022-04-07 11:02:18.413 [utbot-ZZR7      ]       BaseForkTask.cpp:77     ERR| .   Exit status: 2
2022-04-07 11:02:18.414 [utbot-ZZR7      ]       BaseForkTask.cpp:78     ERR| .   See details in "/home/utbot/logs/utbot-ZZR7/coreutils/2022-04-07T07-02-18262.log"
2022-04-07 11:02:18.414 [utbot-ZZR7      ]             Linker.cpp:502    ERR| .   link with stubs failed: env GTEST_FLAGS="" make -j16 -s -f /home/utbot/tmp/utbot-ZZR7/coreutils/GenerationLinkMakefile.mk all
```
- See details
```
In file included from /home/utbot/projects/coreutils/utbot_tests/stubs/src/wc_stub.c:12:
/home/utbot/projects/coreutils/utbot_tests/stubs/src/wc_stub.h:1114:5: error: unknown type name 'wchar_t'
    wchar_t wc;
```

--- a/server/src/clang-utils/SourceToHeaderMatchCallback.cpp
+++ b/server/src/clang-utils/SourceToHeaderMatchCallback.cpp
@@ -144,7 +144,7 @@ void SourceToHeaderMatchCallback::handleTypedef(const TypedefDecl *decl) {
     auto canonicalType = decl->getUnderlyingType().getCanonicalType();
     auto name = decl->getName().str();
     auto canonicalName = canonicalType.getAsString();
-    if (name == "wchar_t") {
+    if (name == "wchar_t" && !forStubHeader) {
         // wchar_t is builtin type in C++ but is typedef in C
         return;
     }

